File: scripts/loc2act/mturk/delete_hits.rb
#!/usr/bin/env ruby

require 'docopt'

require_relative "rturk_hit"

doc = <<DOCOPT
Delete HITs.

Usage:
  #{__FILE__} [--all] [-p | --production]
  #{__FILE__} -h | --help
  #{__FILE__} --version

Options:
  -h --help         Show this screen.
  --version         Show the version.

  --all             Delete all HITs.
  -p --production   Delete HITs from production.

DOCOPT

# Delete all approved/rejected hits
def delete
  puts "Deleting all reviewed assignments"

  count = 0
  RTurk::Hit.each do |hit|
    hit.assignments.each do |assignment|
      if assignment.status == 'Approved' ||
         assignment.status == 'Rejected'
        assignment.dispose!
        count += 1

        hit.disable!
      end
    end
  end

  puts "Disposed of #{count} reviewed assignments"
end

def delete_all
  puts "Deleting all assignments"

  RTurk::Hit.each do |hit|
    hit.disable!
  end
end

# Parse arguments and call delete function
begin
  args = Docopt::docopt(doc, version: '0.2')
rescue Docopt::Exit => e
  puts e.message
end

if args
  if args['--production']
    RTurk.setup(AWSAccessKey, AWSSecretKey, :sandbox => false)
  else
    RTurk.setup(AWSAccessKey, AWSSecretKey, :sandbox => true)
  end

  if args['--all']
    delete_all
  else
    delete
  end
end

File: crowdsourcing/scripts/act2loc/mturk/delete_hits.py
#!/usr/bin/env python

"""Delete HITs.

Usage:
    delete_hits.py [(-p | --production)]
    delete_hits.py (-h | --help)
    delete_hits.py --version

Options:
    -h --help           Show this screen.
    --version           Show version.

    -p --production     Publish in production.
"""
from boto.mturk.connection import MTurkConnection
from docopt import docopt

import ConfigParser

def delete_hits(hits):
    for hit in hits:
        mtc.disable_hit(hit.HITId)

if __name__ == '__main__':
    arguments = docopt(__doc__, version='1.0-SNAPSHOT')
    if arguments['--production']:
        HOST = 'mechanicalturk.amazonaws.com'
    else:
        HOST = 'mechanicalturk.sandbox.amazonaws.com'

    config = ConfigParser.ConfigParser()
    config.read('login.ini')
    access_key = config.get('DEFAULT', 'access_key')
    secret_key = config.get('DEFAULT', 'secret_key')

    mtc = MTurkConnection(aws_access_key_id=access_key,
                          aws_secret_access_key=secret_key,
                          host=HOST)
    delete_hits(mtc.get_all_hits())
